Assets/GothicVR/Scripts/UI/TurnSettingDropdownController.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit;
+using GVR.Phoenix.Interface;
+public class TurnSettingDropdownController : MonoBehaviour
+{
+    public GameObject textAsset;
+    public GameObject locomotionsystem;
+    public ActionBasedSnapTurnProvider snapTurn;
+    public ActionBasedContinuousTurnProvider continuousTurn;
+    // Start is called before the first frame update
+    void Awake()
+    {
+        var dropdown = transform.GetComponent<TMP_Dropdown>();
+        dropdown.options.Clear();
+
+        List<string> items = new List<string>();
+        items.Add("Snap turn");
+        items.Add("Continuous turn");
+
+        foreach (var item in items)
+        {
+            dropdown.options.Add(new TMP_Dropdown.OptionData() { text = item });
+
+        }
+        dropdown.itemText.font = GameData.I.GothicSubtitleFont;
+
+        dropdown.onValueChanged.AddListener(delegate { DropdownItemSelected(dropdown); });
+        snapTurn = locomotionsystem.GetComponent<ActionBasedSnapTurnProvider>();
+        continuousTurn = locomotionsystem.GetComponent<ActionBasedContinuousTurnProvider>();
+    }
+
+    void DropdownItemSelected(TMP_Dropdown dropdown)
+    {
+        switch (dropdown.value)
+        {
+            case 0:
```

Comment:
I recommend introducing an enum to make it clear for the reader what to expect as 0/1 is a magic number here.

e.g.

```c#
    private enum SnapTurnSetting
    {
        Snap,
        Continuous
    }
```

You can also use this enum to fill the label itself.
I think this is self explanatory until we start with translations in the future.

```c#
        items.Add(SnapTurnSetting.Snap.ToString());
        items.Add(SnapTurnSetting.Continuous.ToString());
```

Reply:
Is there an easy option to really use eg "Continuous Turn" with a blank space in between? It seems that's not possible for enums. I would like to go for it. Otherwise it looks a little tech like.

Reply:
I found no easy way for the labels. Only with struct but this complicates the usage. Feel free to still do the enum for using it in the end as switch-case. But not labelling it.